src/coreclr/gc/gc.cpp

```diff
@@ -22846,25 +22850,37 @@ void gc_heap::gc1()
 #ifdef DYNAMIC_HEAP_COUNT
 size_t gc_heap::get_total_soh_stable_size()
 {
-    size_t total_stable_size = 0;
-    bool use_max_gen_p = (dynamic_heap_count_data.current_gen2_samples_count > 0);
-    int gen_number = max_generation;
-    for (int i = 0; i < gc_heap::n_heaps; i++)
+    if (current_total_soh_stable_size)
     {
-        gc_heap* hp = g_heaps[i];
-
-        if (use_max_gen_p)
-        {
-            dynamic_data* dd = hp->dynamic_data_of (gen_number);
-            total_stable_size += dd_current_size (dd) + dd_desired_allocation (dd);
-        }
-        else
+        return current_total_soh_stable_size;
+    }
+    else
```

Comment:
style - the `else` is unnecessary since the `if` can't fall-through (question of `if/else` symmetry vs less indentation for the bulk of the method)

Reply:
not sure what you meant by else is unnecessary. else is needed because current_total_soh_stable_size starts as 0.

Reply:
I mean the `else` itself, not the code in the block.

`if (b) return; else foo();` is the same as `if (b) return; foo();`

Reply:
I would actually much prefer to write the else block explicitly - it makes it very clear that there are 2 cases.

Reply:
that's fair.  This code shape can be read as 2 cases (as you are, if/else makes sense) or "early exit case, then do the normal thing" (if without else makes sense)